### PR TITLE
Nullable System.Diagnostics.Store

### PR DESCRIPTION
**Pull request by Anipik**

### CODE REVIEW

**Comment by stephentoub** [comment]
It's hard to review an internal interface on its own.  Can you please include also types that implement the interface?

File: src/System.Private.CoreLib/shared/System/Diagnostics/SymbolStore/ISymbolDocumentWriter.cs [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
 
+#nullable enable
 namespace System.Diagnostics.SymbolStore
 {
     public interface ISymbolDocumentWriter
     {
-        void SetCheckSum(Guid algorithmId, byte[] checkSum);
-        void SetSource(byte[] source);
+        void SetCheckSum(Guid algorithmId, byte?[]? checkSum);
+        void SetSource(byte?[]? source);
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Diagnostics/SymbolStore/ISymbolDocumentWriter.cs]: Please revert the `byte?` back to `byte` for all such cases.  There's actually a huge difference between `byte?` and `byte`... the former is actually `Nullable<byte>`.

File: src/System.Private.CoreLib/shared/System/Diagnostics/SymbolStore/ISymbolDocumentWriter.cs [hunk 2]
```diff
@@ -2,11 +2,12 @@
...
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
 namespace System.Diagnostics.SymbolStore
 {
     public interface ISymbolDocumentWriter
     {
-        void SetCheckSum(Guid algorithmId, byte[] checkSum);
-        void SetSource(byte[] source);
+        void SetCheckSum(Guid algorithmId, byte?[]? checkSum);
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Diagnostics/SymbolStore/ISymbolDocumentWriter.cs]: Why is this nullable?  Are there implementations that expect null?  The one you included in this PR will null ref is null is passed for `checkSum`.

File: src/System.Private.CoreLib/shared/System/Diagnostics/SymbolStore/ISymbolDocumentWriter.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Diagnostics/SymbolStore/ISymbolDocumentWriter.cs]: Similar question; why is this nullable?

File: src/System.Private.CoreLib/src/System/Diagnostics/SymbolStore/ISymWriter.cs [hunk 3]
```diff
@@ -26,7 +24,7 @@ internal interface ISymbolWriter
         // Define a source document. Guid's will be provided for the
         // languages, vendors, and document types that we currently know
         // about.
-        ISymbolDocumentWriter DefineDocument(string url,
+        ISymbolDocumentWriter? DefineDocument(string url,
```
Comment (stephentoub) [inline src/System.Private.CoreLib/src/System/Diagnostics/SymbolStore/ISymWriter.cs]: Implementations of this return null?

Comment (Anipik) [reply src/System.Private.CoreLib/src/System/Diagnostics/SymbolStore/ISymWriter.cs]: yes here https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/Reflection/Emit/ISymWrapperCore.cs#L182

**Review by stephentoub — APPROVED** [review]

**Review by safern — COMMENTED** [review]
LGTM

**Pull request merged**